framework/db/src/main/java/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -2055,7 +2046,17 @@ public Integer getCount(SearchCriteria<T> sc) {
 
     @DB()
     protected StringBuilder createCountSelect(SearchCriteria<?> sc, final boolean whereClause) {
-        StringBuilder sql = new StringBuilder(_count);
+        if (sc == null)
```

Comment:
This can potentially cause NPE, @anuragaw can you check who are the users of the method and potentially what must be the ideal return value? The `createCountSelect` and its indirect users have atleast 70 usages codebase-wide, therefore while fixing the bug, new regression should be avoided.